src/main.ts

```diff
@@ -45,10 +46,10 @@ async function run(): Promise<void> {
     if (isGraalVMforJDK17OrLater) {
       switch (distribution) {
         case c.DISTRIBUTION_GRAALVM:
-          graalVMHome = await graalvm.setUpGraalVMJDK(javaVersion)
+          graalVMHome = await graalvm.setUpGraalVMJDK(javaVersion, customDistributionUrl)
```

Comment:
Instead of letting `customDistributionUrl` creep into all these setUp functions, I'd prefer if we handle ` customDistributionUrl.length > 0` in one place. What do you think?